modules/wallet/transactionbuilder.go

```diff
@@ -634,3 +634,127 @@ func (w *Wallet) RegisterTransaction(t types.Transaction, parents []types.Transa
 func (w *Wallet) StartTransaction() modules.TransactionBuilder {
 	return w.RegisterTransaction(types.Transaction{}, nil)
 }
+
+// SpendableOutputs returns the outputs spendable by the wallet.
+func (w *Wallet) SpendableOutputs() []modules.SpendableOutput {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	// ensure durability of reported outputs
+	w.syncDB()
+
+	var outputs []modules.SpendableOutput
+	dbForEachSiacoinOutput(w.dbTx, func(scoid types.SiacoinOutputID, sco types.SiacoinOutput) {
+		outputs = append(outputs, modules.SpendableOutput{
+			FundType:   types.SpecifierSiacoinOutput,
+			ID:         types.OutputID(scoid),
+			UnlockHash: sco.UnlockHash,
+			Value:      sco.Value,
+		})
+	})
+	dbForEachSiafundOutput(w.dbTx, func(sfoid types.SiafundOutputID, sfo types.SiafundOutput) {
+		outputs = append(outputs, modules.SpendableOutput{
+			FundType:   types.SpecifierSiafundOutput,
+			ID:         types.OutputID(sfoid),
+			UnlockHash: sfo.UnlockHash,
+			Value:      sfo.Value,
+		})
+	})
+
+	// set the confirmation height for each output
+outer:
+	for i, o := range outputs {
+		txnIndices, _ := dbGetAddrTransactions(w.dbTx, o.UnlockHash)
+		for _, j := range txnIndices {
+			pt, err := dbGetProcessedTransaction(w.dbTx, j)
+			if err != nil {
+				continue
+			}
+			for _, sco := range pt.Outputs {
+				if sco.ID == o.ID {
+					outputs[i].ConfirmationHeight = pt.ConfirmationHeight
+					continue outer
+				}
+			}
+		}
+	}
+
+	return outputs
+}
+
+// SignTransaction signs txn using secret keys known to the wallet. toSign
+// maps the ParentID of each unsigned input to the UnlockHash of that input's
+// desired UnlockConditions. SignTransaction fills in the UnlockConditions for
+// each such input and adds a corresponding signature.
+func (w *Wallet) SignTransaction(txn *types.Transaction, toSign map[types.OutputID]types.UnlockHash) error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if !w.unlocked {
+		return modules.ErrLockedWallet
+	}
+	return signTransaction(txn, w.keys, toSign)
+}
+
+// SignTransaction signs txn using secret keys derived from seed. toSign maps
+// the ParentID of each unsigned input to the UnlockHash of that input's
+// desired UnlockConditions. SignTransaction fills in the UnlockConditions for
+// each such input and adds a corresponding signature.
+//
+// SignTransaction must derive all of the keys from scratch, so it is
+// appreciably slower than calling the Wallet.SignTransaction method. Only the
+// first 1 million keys are derived.
+func SignTransaction(txn *types.Transaction, seed modules.Seed, toSign map[types.OutputID]types.UnlockHash) error {
```

Comment:
Would it makes sense to create a standalone utility too? Something like `siasign` instead of `siac` that keeps the keys in memory once generated and can be used to rapidly sign transactions without having to have a full locked full node installed?

Reply:
if someone wants to, it certainly wouldn't be very difficult. In practice I think most people will just use `siac`.

Reply:
at one point we had siag which was essentially a key generator. I extended it to support signing and multisig as well, but I don't think that we ever released that extension.

I think having `siac` do everything is fine.